### PR TITLE
chore: add error handling

### PR DESCRIPTION
**Pull request by DaoDev44**
- Add error handling to `buildSendTransaction`

### CODE REVIEW

File: packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts [hunk 1]
```diff
@@ -98,14 +98,22 @@ export class ChainAdapter
       const account = await this.getAccount(from)
 
       if (sendMax) {
-        tx.value = bnOrZero(account.balance).minus(gas).toString()
+        try {
+          const val = bnOrZero(account.balance).minus(gas)
+          if (!isFinite(val.toNumber()) || val.toNumber() < 0) {
```
Comment (0xdef1cafe) [inline packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts:103]: you can use val.isFinite() and val.lte(0) here

File: packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts [hunk 2]
```diff
@@ -98,14 +98,22 @@ export class ChainAdapter
...
+          return ErrorHandler(error)
+        }
       }
 
       const utx: CosmosTx = {
         fee: {
           amount: [
             {
-              amount: bnOrZero(fee).toString(),
+              amount: String(bnOrZero(fee)),
```
Comment (0xdef1cafe) [inline packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts:116]: shouldn't we be calling .toString() with some precision stuff instead?

Comment (DaoDev44) [reply packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts:116]: The precision is handled on the frontend I believe, so I don't think we need to specify that. The `String` vs `toString` can go either way for me.

File: packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts [hunk 3]
```diff
@@ -98,14 +98,22 @@ export class ChainAdapter
       const account = await this.getAccount(from)
 
       if (sendMax) {
-        tx.value = bnOrZero(account.balance).minus(gas).toString()
+        try {
+          const val = bnOrZero(account.balance).minus(gas)
+          if (val.isFinite() || val.lte(0)) {
+            throw new Error('CosmosChainAdapter: transaction value is invalid')
```
Comment (gomesalexandre) [inline packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts:104]: Shouldn't we log the actual wrong tx value?
```suggestion
            throw new Error(`CosmosChainAdapter: transaction value is invalid: ${val.toString()}`)
```

File: packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts [hunk 4]
```diff
@@ -134,8 +142,8 @@ export class ChainAdapter
         addressNList,
         tx: utx,
         chain_id: caip2.ChainReference.CosmosHubMainnet,
-        account_number: account.chainSpecific.accountNumber,
-        sequence: account.chainSpecific.sequence
+        account_number: String(account.chainSpecific.accountNumber),
+        sequence: String(account.chainSpecific.sequence)
```
Comment (gomesalexandre) [inline packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts:146]: According to typings, these should already come as a string so no need to cast them

**Review by gomesalexandre — APPROVED** [review]
One small comment regarding what looks like a casting we don't need, otherwise looks good

**Review by gomesalexandre — APPROVED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**